src/screens/IndexScreen.js

```diff
@@ -367,6 +369,40 @@ export const IndexScreen = ({ navigation, route }) => {
 
   const fetchPolicy = graphqlFetchPolicy({ isConnected, isMainserverUp });
 
+  useLayoutEffect(() => {
+    if (query === QUERY_TYPES.POINTS_OF_INTEREST && showMap) {
+      navigation.setOptions({
+        headerLeft: () => (
+          <HeaderLeft
+            onPress={() => {
+              const selectedFilter = filterType.find((entry) => entry.selected);
+              const updatedFilter = filterType.map((entry) => {
+                if (entry.id !== selectedFilter.id) {
+                  entry.selected = true;
+                } else {
+                  entry.selected = false;
+                }
+                return entry;
+              });
+
+              setFilterType(updatedFilter);
+            }}
+            backImage={
+              <Icon.Close
+                color={colors.lightestText}
+                style={{ paddingHorizontal: normalize(14) }}
+              />
+            }
+          />
+        )
+      });
+    } else {
+      navigation.setOptions({
+        headerLeft: () => <HeaderLeft onPress={() => navigation.goBack()} />
```

Comment:
sadly this effect is running all the time on index screens now. but ok, with having our index screen refactored soon, we will have this logic only remaining for the index of points of interest.